nilearn/datasets/struct.py

```diff
@@ -106,55 +119,348 @@ def fetch_icbm152_2009(data_dir=None, url=None, resume=True, verbose=1):
     return Bunch(**params)
 
 
-def load_mni152_template():
-    """Load skullstripped 2mm version of the MNI152 originally distributed
-    with FSL.
+def load_mni152_template(resolution=None):
```

Comment:
due to the need to rescale the template, the loader now takes over 1s (was around .01 before) for the template or the mask

```suggestion
@functools.lru_cache
def load_mni152_template(resolution=None):
```

caching it will allow only paying this price once.
you can add it just for the whole brain template as this template and mask is used all the time (default for masking, plotting, etc), the wm and gm templates may not need it because they are used less often.
(you will need to `import functools` from the standard library)

you will also need to add `loader.cache_clear()` after defining `loader` in the test suggested in [this comment](https://github.com/nilearn/nilearn/pull/2738#issuecomment-869168559)

Reply:
Ok. But this last snippet of code is already written in the function, exactly after defining the `loader`, no?